src/libraries/Common/tests/System/Net/Prerequisites/NetCoreServer/Handlers/VerifyUploadHandler.cs

```diff
@@ -35,31 +35,40 @@ public static async Task InvokeAsync(HttpContext context)
             // Check MD5 checksum for non-empty request body.
             if (requestBodyBytes.Length > 0)
             {
-                // Get expected MD5 hash of request body.
-                string expectedHash = context.Request.Headers["Content-MD5"];
-                if (string.IsNullOrEmpty(expectedHash))
-                {
-                    context.Response.StatusCode = 400;
-                    context.Response.SetStatusDescription("Missing 'Content-MD5' request header");
-                    return;
-                }
-
-                // Compute MD5 hash of received request body.
-                string actualHash;
-                using (MD5 md5 = MD5.Create())
-                {
-                    byte[] hash = md5.ComputeHash(requestBodyBytes);
-                    actualHash = Convert.ToBase64String(hash);
-                }
-
-                if (expectedHash == actualHash)
+                string skip = context.Request.Headers["Content-MD5-Skip"];
+                if(!string.IsNullOrEmpty(skip)) 
                 {
                     context.Response.StatusCode = 200;
                 }
                 else
                 {
```

Comment:
We could get rid of one level of nesting:
```suggestion
                if(!string.IsNullOrEmpty(skip)) 
                {
                    context.Response.StatusCode = 200;
                    return;
                }
```
But this is thing of taste, so feel free to disregard, it's perfectly fine as-is as well.